src/paho/mqtt/client.py

```diff
@@ -2723,6 +2737,8 @@ def _buffered_read(self, length):
 
     def _recv_impl(self, length):
 
+        payload = ""
```

Comment:
This is fixed in #130 (merged in fixes not develop). But #130 use bytearray(). IIRC this is the correct type for payload.
Could you either don't add this in the PR or use the same type.